service/frontend/workflowHandler.go

```diff
@@ -3399,3 +3392,42 @@ func (wh *WorkflowHandler) makeFakeContinuedAsNewEvent(
 		},
 	}, nil
 }
+
+func (wh *WorkflowHandler) validateNamespace(
+	str string,
+) error {
+	if str == "" {
+		return errNamespaceNotSet
+	}
+	if err := wh.validateUTF8String(str); err != nil {
+		return err
+	}
+	if len(str) > wh.config.MaxIDLengthLimit() {
+		return errNamespaceTooLong
+	}
+	return nil
+}
+
+func (wh *WorkflowHandler) validateWorkflowID(
+	str string,
+) error {
+	if str == "" {
+		return errWorkflowIDNotSet
+	}
+	if err := wh.validateUTF8String(str); err != nil {
+		return err
+	}
+	if len(str) > wh.config.MaxIDLengthLimit() {
+		return errWorkflowIDTooLong
+	}
+	return nil
+}
+
+func (wh *WorkflowHandler) validateUTF8String(
```

Comment:
can you have a unit test just to verify that \ufffd would be caught by this check.

Reply:
sample

```go
hexStr := "ff"
hexBytes, err := hex.DecodeString(hexStr)
s.NoError(err)
s.False(utf8.ValidString(string(hexBytes)))

fmt.Println("#######")
    prettyPrint := func(input interface{}) string {
    binary, _ := json.MarshalIndent(input, "", "  ")
    return string(binary)
}
fmt.Printf("%v\n", prettyPrint(string(hexBytes)))
fmt.Println("#######")
```

output 
```
#######
"\ufffd"
#######
```